### ISSUE TITLE
Error occurred when executing SeargeSDXLSampler2:mat1 and mat2 shapes cannot be multiplied (308x2048 and 1280x768)

### ISSUE THREAD

**Issue by whmc76** [SeargeDP/SeargeSDXL#33]
Error occurred when executing SeargeSDXLSampler2:

mat1 and mat2 shapes cannot be multiplied (308x2048 and 1280x768)

File "E:\ComfyUI_windows_portable\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "E:\ComfyUI_windows_portable\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "E:\ComfyUI_windows_portable\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "E:\ComfyUI_windows_portable\ComfyUI\custom_nodes\SeargeSDXL\modules\sampling.py", line 192, in sample
precondition_result = nodes.common_ksampler(refiner_model, noise_seed + 2, steps, cfg, sampler_name, scheduler, refiner_positive, refiner_negative, latent_image, denoise=denoise, disable_noise=False, start_step=steps - refiner_prep_steps, last_step=steps, force_full_denoise=False)
File "E:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1176, in common_ksampler
samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\sample.py", line 88, in sample
samples = sampler.sample(noise, positive_copy, negative_copy, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 716, in sample
samples = getattr(k_diffusion_sampling, "sample_{}".format(self.sampler))(self.model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 613, in sample_dpmpp_2m_sde
denoised = model(x, sigmas[i] * s_in, **extra_args)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 319, in forward
out = self.inner_model(x, sigma, cond=cond, uncond=uncond, cond_scale=cond_scale, cond_concat=cond_concat, model_options=model_options, seed=seed)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\external.py", line 125, in forward
eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\external.py", line 151, in get_eps
return self.inner_model.apply_model(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 307, in apply_model
out = sampling_function(self.inner_model.apply_model, x, timestep, uncond, cond, cond_scale, cond_concat, model_options=model_options, seed=seed)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 285, in sampling_function
cond, uncond = calc_cond_uncond_batch(model_function, cond, uncond, x, timestep, max_total_area, cond_concat, model_options)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 262, in calc_cond_uncond_batch
output = model_function(input_x, timestep_, **c).chunk(batch_chunks)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\model_base.py", line 61, in apply_model
return self.diffusion_model(xc, t, context=context, y=c_adm, control=control, transformer_options=transformer_options).float()
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 620, in forward
h = forward_timestep_embed(module, h, emb, context, transformer_options)
File "E:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-animatediff\animatediff\nodes.py", line 37, in forward_timestep_embed
x = layer(x, context, transformer_options)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\attention.py", line 695, in forward
x = block(x, context=context[i], transformer_options=transformer_options)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\attention.py", line 527, in forward
return checkpoint(self._forward, (x, context, transformer_options), self.parameters(), self.checkpoint)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 123, in checkpoint
return func(*inputs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\attention.py", line 627, in _forward
n = self.attn2(n, context=context_attn2, value=value_attn2)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\attention.py", line 422, in forward
k = self.to_k(context)
File "E:\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI_windows_portable\ComfyUI\comfy\ops.py", line 18, in forward
return torch.nn.functional.linear(input, self.weight, self.bias)

**Comment by SeargeDP** [comment]
Are you using different models than the SDXL base and refiner model?

**Comment by mirek190** [comment]
update comfy ui ... I had similar problem

**Comment by whmc76** [comment]
yes, i want to use some other SDXL models download from civitai.
and I've tested several different models, both the base model and the sdxl model made by others, to reproduce this problem. etc:dreamshaperXL alpha2

**Comment by SeargeDP** [comment]
And did you make sure Comfy is updated to the latest version as mirek190 suggested?

**Comment by whmc76** [comment]
yes, I'm used to updating the version and plugins when I open program every day

**Comment by SeargeDP** [comment]
Looking at the stack trace again, I'm wondering if you loaded the right model as refiner model. It's an error message that reminds me of an issue I had once when I accidentally loaded the base model as the refiner.

**Comment by SeargeDP** [comment]
Is this still an issue?

**Comment by whmc76** [comment]
![image](https://github.com/SeargeDP/SeargeSDXL/assets/129386342/07ceab96-5a2f-4c42-a537-381072d99577)

**Comment by SeargeDP** [comment]
Thanks for posting the image, now I can see the problem. You encoded the prompts with the base CLIP and then feed them into the positive and negative inputs. But you need to encode the prompts for the refiner with the refiner CLIP.
So you should duplicate the CLIP Text Encode nodes you have, feed the 2 new ones with the refiner CLIP, and then connect those conditionings to the refiner_positive and refiner_negative inputs on the sampler.
